lib/Zonemaster/Backend/DB.pm

```diff
@@ -761,6 +790,50 @@ sub to_iso8601 {
     return $time;
 }
 
+sub add_result_entry {
+    my ( $self, $hash_id, $entry ) = @_;
+
+    my $json = JSON::PP->new->allow_blessed->convert_blessed->canonical;
+
+    my $nb_inserted = $self->dbh->do(
+        "INSERT INTO result_entries (hash_id, level, module, testcase, tag, timestamp, args) VALUES (?, ?, ?, ?, ?, ?, ?)",
+        undef,
+        $hash_id,
+        $entry->{level},
+        $entry->{module},
+        $entry->{testcase},
+        $entry->{tag},
+        $entry->{timestamp},
+        $json->encode( $entry->{args} ),
+    );
+    return $nb_inserted;
+}
+
+sub add_result_entries {
+    my ( $self, $hash_id, $entries ) = @_;
+    my @records;
+
+    my $json = JSON::PP->new->allow_blessed->convert_blessed->canonical;
+
+    foreach my $m ( @$entries ) {
+        my $r = [
+            $hash_id,
+            $m->level,
+            $m->module,
+            $m->testcase,
+            $m->tag,
+            $m->timestamp,
+            $json->encode( $m->args // {} ),
+        ];
+
+        push @records, $r;
+    }
+    my $query_values = join ", ", ("(?, ?, ?, ?, ?, ?, ?)") x @records;
+    my $query = "INSERT INTO result_entries (hash_id, level, module, testcase, tag, timestamp, args) VALUES $query_values";
+    my $sth = $self->dbh->prepare($query);
+    $sth = $sth->execute(map { @$_ } @records);
+}
```

Comment:
These two methods, `add_result_entry()` and `add_result_entries()` do the same thing, as far as I can see. Just that the first does it one entry and the other on multiple. Is there a reason for not using just one method, e.g. requiring "entry" to always be a hash reference even if there is only one member?

Reply:
Thanks for the review and suggestion. I've updated the code to make it variadic as suggested.